src/ucp/wireup/wireup_ep.c

```diff
@@ -589,13 +613,33 @@ int ucp_wireup_aux_ep_is_owner(ucp_wireup_ep_t *wireup_ep, uct_ep_h owned_ep)
 int ucp_wireup_ep_is_owner(uct_ep_h uct_ep, uct_ep_h owned_ep)
 {
     ucp_wireup_ep_t *wireup_ep = ucp_wireup_ep(uct_ep);
+    ucp_ep_h tmp_ep;
+    ucp_lane_index_t lane;
 
     if (wireup_ep == NULL) {
         return 0;
     }
 
-    return (ucp_wireup_aux_ep_is_owner(wireup_ep, owned_ep)) ||
-           (wireup_ep->super.uct_ep == owned_ep);
+    if ((ucp_wireup_aux_ep_is_owner(wireup_ep, owned_ep)) ||
+        (wireup_ep->super.uct_ep == owned_ep)) {
+        return 1;
+    }
+
+    tmp_ep = wireup_ep->tmp_ep;
+    if (tmp_ep != NULL) {
+        for (lane = 0; lane < ucp_ep_num_lanes(tmp_ep); ++lane) {
+            if (lane == ucp_ep_get_cm_lane(tmp_ep)) {
+                continue;
+            }
+
+            if ((uct_ep == tmp_ep->uct_eps[lane]) ||
+                ucp_wireup_ep_is_owner(tmp_ep->uct_eps[lane], uct_ep)) {
+                return 1;
+            }
+        }
```

Comment:
can we unite common code with worker's err handling function?

Reply:
done